src/torchcodec/_samplers/video_clip_sampler.py

```diff
@@ -212,7 +212,7 @@ def _get_clips_for_index_based_sampling(
         sample_start_index = max(0, index_based_sampler_args.sample_start_index)
         sample_end_index = (
             min(
-                index_based_sampler_args.sample_end_index,
+                index_based_sampler_args.sample_end_index + 1,
                 metadata_json["numFrames"],
```

Comment:
I am wondering if this needs to have numFrames - 1 here, because we call linspace on it below which can return a value equal to this value.

So that will also index out of bounds.

Reply:
No, the `numFrames` is actually the correct one. We subtract 1 in the next 2 lines `-index_based_sampler_args.video_frame_dilation * index_based_sampler_args.frames_per_clip`.